### ISSUE TITLE
Simulation mode set via string - unexpected behaviour when setting `simulate` without `data`

### ISSUE THREAD

**Issue by nikbpetrov** [jspsych/jsPsych#2912]
When I create a global `simulation_options` object and define values [using a string](https://www.jspsych.org/7.1/overview/simulation/#experiment-level-options), such as 'long_response' as in the docs, I cannot set only the simulate parameter, without the data one.

More specifically, this will NOT work (no `data` parameter):
```js
const simulation_options = {
  long_response: {
    simulate: true
    }
  }
}
```

this works (some default data parameter that gets merged with the default data):

```js
const simulation_options = {
  long_response: {
    simulate: true,
    data: {}
    }
  }
}
```

and this also works (we don't run a simulation at all):
```js
const simulation_options = {
  long_response: {
    simulate: false
    }
  }
}
```

I expect this to work by using the default data:
```js
const simulation_options = {
  long_response: {
    simulate: true
    }
  }
}
```

And I also expect certain precedence (e.g. maybe I have defined default data in the simulation_options and expect this to be inherited into the 'long_response', see #2911 

P.S. I know I have opened 4-5 issues today on this, and sorry for inundating you, but I am developing a large, custom online experiment and need to take care of a lot of details - using the simulation mode is a blessing!

**Comment by nikbpetrov** [comment]
> > P.S. I know I have opened 4-5 issues today on this, and sorry for inundating you, but I am developing a large, custom online experiment and need to take care of a lot of details - using the simulation mode is a blessing!
> 
> This is great. Thanks for reporting all of these. I'm probably not going to be able to work actively on it for a couple weeks, but I will see if I can squeeze in some fixes.

No worries at all. I have found workarounds for myself. I will try to share a version of my code in the Discussions once the full setup is complete as I really think my code showcases a lot of the strengths of the simulation mode as it is currently set up -- super turbo scalable!

**Comment by bjoluc** [comment]
> I have found workarounds for myself.

@nikbpetrov Great to hear you were able to work around the quirks. As always, I'm curious whether #2858 works as intended. If you could devote some time for testing it against #2911 and #2912, that would be great! Otherwise, I'll look into this soon. Thanks!

**Comment by jodeleeuw** [comment]
Hey @nikbpetrov,

I'm having a hard time reproducing this one. 

I added a test case in https://github.com/jspsych/jsPsych/commit/b6d4b182789698c1d88ac95e096f5feff7545172

Does the test case match the configuration that you are having trouble with?

I did push some fixes for other issues on this branch already, so I suppose it is possible that those fixed this if the bug was due to some convoluted handling of the `simulation_options`.
